routes/plan.js

```diff
@@ -152,54 +187,88 @@ const getStartYear = (year) => {
-      // TODO: delete reviews
-      users
+    .then(async (plan) => {
+      //delete distributions, fineReqs, courses, and years
+      await distributions.deleteMany({ plan_id: plan._id }).exec();
+      await fineRequirements.deleteMany({ plan_id: plan._id }).exec();
+      await courses.deleteMany({ plan_id: plan._id }).exec();
+      await years.deleteMany({ plan_id: plan._id }).exec();
+      //delete plan_id from user
+      await users
         .findByIdAndUpdate(
-          //delete plan_id from user
           plan.user_id,
           { $pull: { plan_ids: plan._id } },
           { new: true, runValidators: true }
         )
         .exec();
-      returnData(plan, res);
+      let deletedPlan = { ...plan._doc };
+      returnData(deletedPlan, res);
     })
     .catch((err) => errorHandler(res, 400, err));
 });
 
 //***need to consider not allow user to change major for a plan ***/
+// updates a plan's major(s) and name
 router.patch("/api/plans/update", (req, res) => {
   const id = req.body.plan_id;
-  const majors = req.body.majors;
+  const major_ids = req.body.major_ids;
   const name = req.body.name;
-  if (!(majors || name)) {
+  if (!(major_ids || name)) {
     errorHandler(res, 400, "Must update majors or name.");
   } else {
     let updateBody = {};
-    if (majors) {
-      updateBody.majors = majors;
+    if (major_ids) {
+      updateBody.major_ids = major_ids;
     }
     if (name) {
       updateBody.name = name;
     }
     plans
       .findByIdAndUpdate(id, updateBody, { new: true, runValidators: true })
-      .then((plan) => {
-        reviews
+      .then(async (plan) => {
+        // add dists for new major, if any
+        await addMajorDistributions(plan);
+        // remove dists and fineReqs for deleted major, if any
+        await distributions.find({ plan_id: plan._id }).then(async (dists) => {
+          for (let dist of dists) {
+            if (!plan._doc.major_ids.includes(dist.major_id)) {
+              // maintain courses array fields
+              await courses.updateMany(
+                { plan_id: id },
+                { $pull: { distribution_ids: dist._id } }
+              ).exec();
+              for (let f_id of dist.fineReq_ids) {
+                await courses.updateMany(
+                  { plan_id: id },
+                  { $pull: { fineReq_ids: f_id } }
+                ).exec();
+              }
+              // delete documents
+              await distributions.findByIdAndDelete(dist._id).exec();
+              await fineRequirements.deleteMany({ distribution_id: dist._id }).exec();
+            }
+          }
+        });
+        // return plan with reviews and distributions
+        const distObjs = await distributions
+          .find({ plan_id: plan._id })
+          .populate("fineReq_ids")
+          .exec();
+        await reviews
           .find({ plan_id: id })
           .populate("reviewer_id")
           .then((revs) => {
```

Comment:
is "revs" reviewer object or actual reviews? you can use the same variable name as the field name "reviewers"